frontend/src/reducer/publicationList.js

```diff
@@ -35,14 +36,18 @@ export default function publicationList (state = defaultPublicationList, action)
     case `${UPDATE}/${WORKSPACE_PUBLICATION_LIST}`:
       return uniqByContentId(sortByModifiedDate(state))
 
-    case `${UPDATE}/${PUBLICATION}`:
-      return uniqByContentId(state.map(publication => action.publication.content_id === publication.id
+    case `${UPDATE}/${PUBLICATION}`: {
+      const serializedPublication = action.publication.content_id
+        ? serialize(action.publication, serializeContentProps)
+        : action.publication
```

Comment:
what's the purpose of this test?

Reply:
test if the publication is serialized or not